Most of the test failures are fixed in #815 (I'm still flogging that one). 

The new tests are failing as expected:

```
>       assert image_name in result_stdout
E       AssertionError: assert 'quay.io/kdelee/does-not-exist' in 'The command was not found or was not executable: podman.\n'
```

Unfortunately switching to using `docker` won't work since this change uses a flag, `--authfile`, that `docker` does not support.

This puts us in the uncomfortable situation of needing to 1) improve CI to allow running `podman` or 2) skip these tests and merge code we are not testing. I am not a fan of option two, especially if this code will be in a supported product.

I do plan to add `podman` support to our CI, but I don't have time to work on that currently.

--- a/ansible_runner/config/_base.py
+++ b/ansible_runner/config/_base.py
@@ -451,7 +451,7 @@ class BaseConfig(object):
 
         # For run() and run_async() API value of base execution_mode is 'BaseExecutionMode.NONE'
         # and the container volume mounts are handled separately using 'container_volume_mounts'
-        # hence ignore additonal mount here
+        # hence ignore additional mount here
         if execution_mode != BaseExecutionMode.NONE:
             if execution_mode == BaseExecutionMode.ANSIBLE_COMMANDS:
                 self._handle_ansible_cmd_options_bind_mounts(new_args, cmdline_args)
@@ -532,7 +532,7 @@ class BaseConfig(object):
 
     def _generate_container_auth_file(self, host, username, password):
         token = "{}:{}".format(username, password)
-        encrypted_container_auth_data = {'auths': {host: {'auth': b64encode(token.encode('UTF-8')).decode('UTF-8')}}}
+        encoded_container_auth_data = {'auths': {host: {'auth': b64encode(token.encode('UTF-8')).decode('UTF-8')}}}
         # Create a new temp file with container auth data
         path = tempfile.mkdtemp(prefix='ansible_runner_registry_%s_' % self.ident)
 
@@ -542,7 +542,7 @@ class BaseConfig(object):
         registry_auth_path = os.path.join(path, 'auth.json')
         with open(registry_auth_path, 'w') as authfile:
             os.chmod(authfile.name, stat.S_IRUSR | stat.S_IWUSR)
-            authfile.write(json.dumps(encrypted_container_auth_data, indent=4))
+            authfile.write(json.dumps(encoded_container_auth_data, indent=4))
         return authfile.name
 
     def wrap_args_with_ssh_agent(self, args, ssh_key_path, ssh_auth_sock=None, silence_ssh_add=False):

--- a/test/integration/containerized/test_container_management.py
+++ b/test/integration/containerized/test_container_management.py
@@ -3,6 +3,7 @@ import shutil
 import time
 import json
 from glob import glob
+from uuid import uuid4
 
 import pytest
 
@@ -127,7 +128,10 @@ def test_registry_auth_file_cleanup(tmp_path, cli):
     with open(os.path.join(private_data_dir, 'env', 'settings'), 'w') as f:
         f.write(json.dumps(settings_data, indent=2))
 
-    cli(['run', private_data_dir, '-p', 'ping.yml'], check=False)
+    this_ident = str(uuid4())[:5]
 
-    registry_files_after = set(glob(auth_registry_glob))
-    assert registry_files_after == registry_files_before
+    cli(['run', private_data_dir, '--ident', this_ident, '-p', 'ping.yml'], check=False)
+
+    discovered_registry_files = set(glob(auth_registry_glob)) - registry_files_before
+    for file_name in discovered_registry_files:
+        assert this_ident not in file_name

--- a/test/integration/containerized/test_container_management.py
+++ b/test/integration/containerized/test_container_management.py
@@ -75,8 +75,9 @@ def test_cancel_will_remove_container(test_data_dir, container_runtime_installed
     ), 'Found a running container, they should have all been stopped'
 
 
+@pytest.mark.skipif(shutil.which('podman') is None, reason='podman not available')
 def test_invalid_registry_host(tmp_path):
-    pdd_path = tmp_path / "private_data_dir"
+    pdd_path = tmp_path / 'private_data_dir'
     pdd_path.mkdir()
     private_data_dir = str(pdd_path)
 
@@ -109,7 +110,7 @@ def test_invalid_registry_host(tmp_path):
 
 
 def test_registry_auth_file_cleanup(tmp_path, cli):
-    pdd_path = tmp_path / "private_data_dir"
+    pdd_path = tmp_path / 'private_data_dir'
     pdd_path.mkdir()
     private_data_dir = str(pdd_path)
 

--- a/test/integration/containerized/test_container_management.py
+++ b/test/integration/containerized/test_container_management.py
@@ -1,5 +1,6 @@
 import os
 import shutil
+import tempfile
 import time
 
 import pytest
@@ -70,3 +71,23 @@ def test_cancel_will_remove_container(test_data_dir, container_runtime_installed
     assert not is_running(
         cli, container_runtime_installed, 'ansible_runner_foo_bar'
     ), 'Found a running container, they should have all been stopped'
+
+
+def test_invalid_registry_host():
+    private_data_dir = tempfile.mkdtemp()
+    res = run(
+        private_data_dir=private_data_dir,
+        playbook='ping.yml',
+        settings={
+            'process_isolation_executable': 'podman',
+            'process_isolation': True,
+            'container_image': 'quay.io/kdelee/awx-ee',
+            'container_options': ['--user=root', '--pull=always'],
+        },
+        container_auth_data={'host': 'https://somedomain.invalid'},
+        ident='awx_123'
+    )
+    assert res.status == 'failed'
+    assert res.rc == 125  # This return code indicates a failed podman run
+
+    shutil.rmtree(private_data_dir)

